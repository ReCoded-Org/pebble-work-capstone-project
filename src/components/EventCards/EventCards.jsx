import Image from "next/image";
import Link from "next/link";
import { useTranslation } from "next-i18next";

import useAuth from "@/hooks/useAuth";

import Button from "@/components/Button";

const EventCards = ({ events = [], isJoined = {}, handleJoinClick }) => {
    const { t } = useTranslation("common");
    const { auth } = useAuth();

    return (
        <div>
            {events.map((event) => (
                <Link href={`/${event._id}`} key={event._id}>
                    <div
                        key={event._id}
                        className='mx-6 mb-4 rounded-md border shadow-lg sm:mx-4'
                    >
                        <div className='mx-4 mt-4 flex flex-col text-center sm:flex-row sm:justify-between'>
                            <h2 className='mb-2 sm:mb-0'>
                                {`${event.date.split("T")[0]}, ${event.date
                                    .split("T")[1]
                                    .slice(0, 8)}`}{" "}
                                {t("eventsPage.eventCards.in")}{" "}
                                {event.address.city}
                            </h2>
                            <div className='flex flex-col justify-between sm:flex-row sm:gap-2'>
                                <div className='flex justify-center pl-3'>
                                    {event.confirmedVolunteers.length === 0 ? (
                                        <h1>
                                            {t(
                                                "eventsPage.eventCards.noAttendeesYet"
                                            )}
                                        </h1>
                                    ) : (
                                        event.confirmedVolunteers
                                            .slice(0, 4)
                                            .map((volunteer) => (
                                                <div
                                                    key={volunteer._id}
                                                    className='-ml-3'
                                                >
                                                    <Image
                                                        className='rounded-full'
                                                        src={
                                                            volunteer.profileImage
                                                        }
                                                        alt='user-images'
                                                        width={25}
                                                        height={25}
                                                    />
                                                </div>
                                            ))
                                    )}
                                </div>
                                {event.confirmedVolunteers.length ===
                                0 ? null : event.confirmedVolunteers.length ===
                                  1 ? (
                                    <h1>
                                        {t(
                                            "eventsPage.eventCards.only1Attendee"
                                        )}
                                    </h1>
                                ) : event.confirmedVolunteers.length >= 2 &&
                                  event.confirmedVolunteers.length <= 3 ? (
                                    <h1>
                                        {" "}
                                        {event.confirmedVolunteers.length}{" "}
                                        {t("eventsPage.eventCards.attendees")}
                                    </h1>
                                ) : event.confirmedVolunteers.length >= 3 ? (
                                    <h1>
                                        + {event.confirmedVolunteers.length - 3}{" "}
                                        {t("eventsPage.eventCards.attendees")}
                                    </h1>
                                ) : null}
                            </div>
                        </div>
                        <div className='m-4 grid-cols-3 content-center gap-4 sm:grid'>
                            <div className='relative col-span-1 mb-4 h-36 overflow-hidden rounded-md border-2 border-black sm:mb-0'>
                                <Image
                                    src={event.coverImage}
                                    alt='event-card-image'
                                    layout='fill'
                                    objectFit='cover'
                                />
                            </div>
                            <div className='col-span-2 flex flex-col justify-between text-center sm:text-left'>
                                <div>
                                    <h1 className='text-2xl font-bold line-clamp-1'>
                                        {event.title}
                                    </h1>
                                    <p className='line-clamp-4 sm:line-clamp-2'>
                                        {event.content}
                                    </p>
                                </div>
                                <div className='mt-2 flex items-end justify-center sm:justify-end '>
                                    {/* if user signed in show join leave buttons */}
                                    {auth?.email && (
                                        <Button
                                            label={
                                                isJoined[event._id]
                                                    ? t(
                                                          "eventsPage.eventCards.leave"
                                                      )
                                                    : t(
                                                          "eventsPage.eventCards.join"
                                                      )
                                            }
                                            textColor='text-white'
                                            bgColor='bg-primary-200'
                                            borderColor='border-primary-200'
                                            height='h-10'
                                            width='w-24'
                                            customStyle='mx-2 my-0'
                                            onClick={(e) => {
                                                e.preventDefault();
                                                handleJoinClick(event._id);
                                            }}
                                        />
                                    )}
                                    {/* if user is NOT signed in show sign in or sign up buttons */}
                                    {!auth?.email && (
                                        <div className='text-blue flex flex-row items-center '>
                                            <Link
                                                className='hover:pointer'
                                                href='/signin'
                                            >
                                                <p className='text-primary-200 underline'>
                                                    {t("common.nav.signIn")}
                                                </p>
                                            </Link>
                                            &nbsp;OR&nbsp;
                                            <Link href='/signup'>
                                                <p className='text-primary-200 underline'>
                                                    {t("common.nav.signUp")}
                                                </p>
                                            </Link>
                                            &nbsp;to join events.
                                        </div>
                                    )}
                                </div>
                            </div>
                        </div>
                    </div>
                </Link>
            ))}
        </div>
    );
};

export default EventCards;
