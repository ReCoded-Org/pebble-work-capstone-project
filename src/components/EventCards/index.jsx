import EventCards from "./EventCards";
export default EventCards;
