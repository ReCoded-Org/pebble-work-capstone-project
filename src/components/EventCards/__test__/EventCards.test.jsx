import renderer from "react-test-renderer";

import EventCards from "../EventCards";

it("renders correctly", () => {
    const tree = renderer.create(<EventCards />).toJSON();
    expect(tree).toMatchSnapshot();
});
