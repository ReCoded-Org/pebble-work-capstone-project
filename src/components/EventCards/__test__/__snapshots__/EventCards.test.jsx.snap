// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="m-5 rounded-md border-2 border-black "
>
  <div
    className="mx-4 mt-4 flex flex-col text-center sm:flex-row sm:justify-between"
  >
    <h2
      className="mb-2 sm:mb-0"
    >
      FRI, JUL -7:00 PM GMT+3
    </h2>
    <div
      className="flex flex-col justify-between sm:flex-row sm:gap-2"
    >
      <div
        className="flex justify-center pl-3"
      >
        <div
          className="-ml-3"
        >
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "inline-block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "overflow": "hidden",
                "padding": 0,
                "position": "relative",
                "width": "initial",
              }
            }
          >
            <span
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            >
              <img
                alt=""
                aria-hidden={true}
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2725%27%20height=%2725%27/%3e"
                style={
                  Object {
                    "background": "none",
                    "border": 0,
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "padding": 0,
                    "width": "initial",
                  }
                }
              />
            </span>
            <img
              alt="user-images"
              data-nimg="intrinsic"
              decoding="async"
              onError={[Function]}
              onLoad={[Function]}
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
            <noscript>
              <img
                alt="user-images"
                data-nimg="intrinsic"
                decoding="async"
                loading="lazy"
                src="/_next/image?url=%2Fimages%2FuserAvatar.jpeg&w=64&q=75"
                srcSet="/_next/image?url=%2Fimages%2FuserAvatar.jpeg&w=32&q=75 1x, /_next/image?url=%2Fimages%2FuserAvatar.jpeg&w=64&q=75 2x"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
            </noscript>
          </span>
        </div>
        <div
          className="-ml-3"
        >
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "inline-block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "overflow": "hidden",
                "padding": 0,
                "position": "relative",
                "width": "initial",
              }
            }
          >
            <span
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            >
              <img
                alt=""
                aria-hidden={true}
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2725%27%20height=%2725%27/%3e"
                style={
                  Object {
                    "background": "none",
                    "border": 0,
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "padding": 0,
                    "width": "initial",
                  }
                }
              />
            </span>
            <img
              alt="user-images"
              data-nimg="intrinsic"
              decoding="async"
              onError={[Function]}
              onLoad={[Function]}
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
            <noscript>
              <img
                alt="user-images"
                data-nimg="intrinsic"
                decoding="async"
                loading="lazy"
                src="/_next/image?url=%2Fimages%2FuserAvatar.jpeg&w=64&q=75"
                srcSet="/_next/image?url=%2Fimages%2FuserAvatar.jpeg&w=32&q=75 1x, /_next/image?url=%2Fimages%2FuserAvatar.jpeg&w=64&q=75 2x"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
            </noscript>
          </span>
        </div>
        <div
          className="-ml-3"
        >
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "inline-block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "overflow": "hidden",
                "padding": 0,
                "position": "relative",
                "width": "initial",
              }
            }
          >
            <span
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            >
              <img
                alt=""
                aria-hidden={true}
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2725%27%20height=%2725%27/%3e"
                style={
                  Object {
                    "background": "none",
                    "border": 0,
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "padding": 0,
                    "width": "initial",
                  }
                }
              />
            </span>
            <img
              alt="user-images"
              data-nimg="intrinsic"
              decoding="async"
              onError={[Function]}
              onLoad={[Function]}
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
            <noscript>
              <img
                alt="user-images"
                data-nimg="intrinsic"
                decoding="async"
                loading="lazy"
                src="/_next/image?url=%2Fimages%2FuserAvatar.jpeg&w=64&q=75"
                srcSet="/_next/image?url=%2Fimages%2FuserAvatar.jpeg&w=32&q=75 1x, /_next/image?url=%2Fimages%2FuserAvatar.jpeg&w=64&q=75 2x"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
            </noscript>
          </span>
        </div>
      </div>
      <p>
        +12 Attendees
      </p>
    </div>
  </div>
  <div
    className="m-4 grid-cols-3 content-center gap-4 sm:grid"
  >
    <div
      className="relative col-span-1 mb-4 h-48 overflow-hidden rounded-md border-2 border-black sm:mb-0"
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": "initial",
            "left": 0,
            "margin": 0,
            "opacity": 1,
            "overflow": "hidden",
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": "initial",
          }
        }
      >
        <img
          alt="event-card-image"
          data-nimg="fill"
          decoding="async"
          onError={[Function]}
          onLoad={[Function]}
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": "cover",
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
        <noscript>
          <img
            alt="event-card-image"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="/_next/image?url=%2Fimages%2Fevent-image.jpg&w=3840&q=75"
            srcSet="/_next/image?url=%2Fimages%2Fevent-image.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fevent-image.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fevent-image.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fevent-image.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fevent-image.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fevent-image.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fevent-image.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fevent-image.jpg&w=3840&q=75 3840w"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": "cover",
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </noscript>
      </span>
    </div>
    <div
      className="col-span-2 flex flex-col justify-between text-center sm:text-left"
    >
      <div>
        <h1
          className="text-2xl font-bold line-clamp-1"
        >
          Title Of The Event
        </h1>
        <p
          className="line-clamp-4"
        >
          Lorem ipsum dolor, sit amet consectetur adipisicing elit. Reprehenderit tempore placeat voluptate officia illo harum enim nulla sunt corporis illum mollitia explicabo ipsam sequi, provident labore perspiciatis tenetur. Magnam sunt, saepe ducimus facere blanditiis possimus tenetur odio error ab esse nobis aperiam alias iusto amet quam, accusantium magni animi eligendi.
        </p>
      </div>
      <div
        className="mt-2 flex items-end justify-center sm:justify-end "
      >
        <button
          className="m-2 w-24 h-10 bg-primary-200  hover:undefined text-white undefined  py-1  px-2 border-primary-200 hover:undefined rounded  hover:shadow-xl"
          onClick={[Function]}
          type="button"
        >
          JOIN
        </button>
      </div>
    </div>
  </div>
</div>
`;
